plugin/evm/validators_normalizer_test.go

```diff
@@ -0,0 +1,66 @@
+package evm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/flare-foundation/coreth/tests/mocks"
+	"github.com/flare-foundation/flare/ids"
+	"github.com/flare-foundation/flare/snow"
+	"github.com/flare-foundation/flare/utils/logging"
+)
+
+func TestValidatorsNormalizer_ByEpoch(t *testing.T) {
+	normalizer, copy := validatorNormalizerWithData(9, 2, true)
+
+	t.Run("nominal case", func(t *testing.T) {
+		for _, i := range []uint64{0, 1} {
+			result, err := normalizer.ByEpoch(i)
+			require.NoError(t, err)
+
+			weighted := normalizer.calcWeightRatio(copy[i])
+			assert.Equal(t, len(weighted), len(result))
+			assert.InDeltaMapValues(t, weighted, result, 0, 0)
+		}
+	})
+
+	t.Run("handles missing key", func(t *testing.T) {
+		_, err := normalizer.ByEpoch(uint64(7))
+		require.NoError(t, err)
+	})
+}
+
+func TestCalcWeightRatio(t *testing.T) {
+	normalizer, _ := validatorNormalizerWithData(9, 2, false)
+
+	validators := mocks.TestValidatorsWeightsData()
+	expected := map[ids.ShortID]uint64{
+		mocks.TestIdsList[2]: 515007957320493220,
+		mocks.TestIdsList[1]: 6465383072520688749,
+		mocks.TestIdsList[0]: 6185718675989118760,
+		mocks.TestIdsList[3]: 9599604527249266409,
+		mocks.TestIdsList[4]: 3417212044008417831,
+	}
+	weighted := normalizer.calcWeightRatio(validators)
+	assert.InDeltaMapValues(t, weighted, expected, 0, 0)
+}
+
+func validatorNormalizerWithData(numValidators, epochs int, duplicateData bool) (*ValidatorsNormalizer, mocks.ValidatorsTestData) {
+	normTestData := mocks.NewValidatorsTestDataByEpochs(numValidators, epochs)
+	copyNormTestData := mocks.ValidatorsTestData{}
+	if duplicateData {
+		copyNormTestData = normTestData.Duplicate()
+	}
+
+	validatorsCache := NewValidatorsCache(normTestData, WithCacheSize(128))
+	for k, v := range normTestData {
+		validatorsCache.cache.Add(k, v)
+	}
+
+	var testCtx = snow.Context{
+		Log: &logging.Log{},
+	}
+	return NewValidatorsNormalizer(testCtx.Log, validatorsCache), copyNormTestData
```

Comment:
```suggestion
	return NewValidatorsNormalizer(&logging.Log{}, validatorsCache), copyNormTestData
```

Not sure what the point is of creating a context here if we only use one of its attributes 🤔

Reply:
Yes, corrected it here https://github.com/flare-foundation/coreth/pull/25/commits/1dd72d6d53018c84ed45547712fe9644c91bf263#diff-3bb410a6d1f259ada3678375ee08a83bd2a8bf2e93945e5c92b92a6f9d99e796R61
Thanks, missed this